reviews/review-01+2020-11-02.md

```diff
@@ -0,0 +1,305 @@
+                work)
+
+        -   3.2, using variables in a script: "Getting the name of the
+            author for only one of the books isn't all that useful. What
+            we really want is a way to get the name of the author from
+            any of our files." (and then go on to talk about variables)
+
+            -   Topic: Using variables
+
+            -   Why: We want to solve the problem of analyzing all of
+                our data.
+
+    -   Examples that could be improved:
+
+        -   2.1, starting to use the shell: I'm looking for something
+            like "Our first shell commands will allow us to navigate and
+            view the contents of our file system, a foundational skill
+            for using the shell to run other commands and use files
+            organized in a project."
+
+        -   3.6, introducing shell variables: I'm missing the final "so
+            what"---like "knowing about shell variables allows us to
+            customize our shell working environment if we use it
+            frequently."
+
+    -   If you only review one chapter for these "topic sentences" I
+        would recommend chapter 7. The content in that chapter is
+        crucially important, and currently can read as somewhat
+        unconnected topics around people and project management. A topic
+        sentence in each section that links back to the goal of
+        empowering people to collaborate together and in a safe and
+        productive manner (or connecting a particular section to a
+        previous section in the chapter, e.g. 7.6 is really a follow on
+        from 7.5) would go a long way towards helping that material
+        cohere. That, or adding more details to the introduction,
+        outlining the major topics that will be covered in the chapter.
+
+-   I had the most difficulty in following chapter 10 (Error Handling);
+    it was the one chapter where I felt like I didn't have a sufficient
+    mental model to understand the details of the chapter. I'm not sure
+    whether the solution is for me to have a better understanding of
+    Python or for the chapter to more explicitly generate a mental model
+    for errors and exceptions.
+
+-   For some of the initial appendices (A, B, C), it wasn't clear if the
+    text was meant to be used as a sample for your own project (or the
+    zipfs sample project) or as a reference for the project that is the
+    manuscript itself.
+
+-   An "if there's time/energy" suggestion: in the final chapter,
+    revisit the profiles from the first chapter, and what chapters of
+    the books were helpful to each person in achieving their goals. E.g.
+    "Amira now has her scripts organized in a repository, following the
+    project scheme described in this book. She made a documentation page
+    for her repository so her co-workers are able to see the work she's
+    done..." etc. Not necessary, but would be fun. :)
```

Comment:
I don't think we need to revisit the profiles, as we've been reading about what Amira and Sami have been doing throughout.

Reply:
I like this suggestion, it might nicely fill out our rather terse finale:
https://github.com/merely-useful/py-rse/blob/e699f2335f46896932eec61ef3122e4295b4b8d3/chapters/finale.Rmd#L7
I'm happy to take a stab at it.